cryptoki/src/session/object_management.rs

```diff
@@ -31,19 +31,7 @@ impl Session {
         let mut object_count = 0;
         let mut objects = Vec::new();
 
-        unsafe {
-            Rv::from(get_pkcs11!(self.client(), C_FindObjects)(
-                self.handle(),
-                object_handles.as_mut_ptr() as CK_OBJECT_HANDLE_PTR,
-                MAX_OBJECT_COUNT.try_into()?,
-                &mut object_count,
-            ))
-            .into_result(Function::FindObjects)?;
-        }
-
-        while object_count > 0 {
-            objects.extend_from_slice(&object_handles[..object_count.try_into()?]);
-
+        loop {
```

Comment:
Good idea to combine both calls in the same loop! I am though generally wary of `loop` though, that in some unexpected data path (now or in a future change) we might end up in an infinite loop.. Is there a way to do the same but having a `while` condition instead? For example by initializing `object_count` to something else than 0?

Reply:
adjusted.